runtime/src/deterministic.rs

```diff
@@ -304,7 +272,7 @@ impl Auditor {
         *hash = hasher.finalize().to_vec();
     }
 
-    fn write_at(&self, partition: &str, name: &[u8], buf: &[u8], offset: u64) {
+    pub(crate) fn write_at(&self, partition: &str, name: &[u8], buf: &[u8], offset: u64) {
```

Comment:
I wonder if we should structure the `auditor` to just expose the underlying `hash` and allow anyone using it to define their own functions over it?

I suppose this could lead to duplicates but we could add a "register" function that asserts there aren't any?

Reply:
That's a reasonable idea, but I think it should probably be in a follow-up PR

Reply:
Works for me. Can you create an issue?